File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/presentation/settings/SpinnerInteractionListener.java
package space.lopatkin.spb.helpboardgamecard.presentation.settings;

import android.view.MotionEvent;
import android.view.View;
import android.widget.AdapterView;

public class SpinnerInteractionListener implements AdapterView.OnItemSelectedListener, View.OnTouchListener {

    private boolean userSelection = false;

    private SettingsViewModel viewModel;

    public SpinnerInteractionListener(SettingsViewModel viewModel) {
        this.viewModel = viewModel;
    }

    @Override
    public boolean onTouch(View v, MotionEvent event) {
        userSelection = true;
        return false;
    }

    @Override
    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
        if (userSelection) {
            Object userChoice = parent.getItemAtPosition(position);
            viewModel.saveKeyboardType(userChoice);
        }
    }

    @Override
    public void onNothingSelected(AdapterView<?> parent) {

    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/presentation/settings/SpinnerInteractionListener.kt
package space.lopatkin.spb.helpboardgamecard.presentation.settings

import android.view.MotionEvent
import android.view.View
import android.widget.AdapterView

class SpinnerInteractionListener : AdapterView.OnItemSelectedListener, View.OnTouchListener {

    private var onEndActionListener: OnEndActionListener? = null
    private var userSelection = false

    override fun onTouch(v: View, event: MotionEvent): Boolean {
        userSelection = true
        return false
    }

    override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {

        if (onEndActionListener != null && userSelection) {
            val userChoice: Any? = parent.getItemAtPosition(position)
            onEndActionListener?.onUserSelection(userChoice)
        }

    }

    override fun onNothingSelected(parent: AdapterView<*>?) {}

    fun setEndActionListener(onEndActionListener: OnEndActionListener) {
        this.onEndActionListener = onEndActionListener
    }

    fun interface OnEndActionListener {

        fun onUserSelection(selection: Any?)

    }

}
